core/templates/services/request-interceptor.service.ts

```diff
@@ -78,15 +78,23 @@ export class RequestInterceptor implements HttpInterceptor {
     if (request.body) {
       return from(this.csrf.getTokenAsync())
         .pipe(
-          switchMap(token => {
+          switchMap((token: string) => {
             if (request.method === 'POST' || request.method === 'PUT') {
-              var body = new FormData();
-              // @ts-ignore
-              body.append('csrf_token', token);
-              body.append('source', document.URL);
-              body.append('payload', JSON.stringify(request.body));
-              // @ts-ignore
-              request.body = body;
+              // If the body of the http request created is already in formData
+              // form, just add the required fields for the request and pass it
+              // on.
+              if (request.body instanceof FormData) {
+                request.body.append('csrf_token', token);
+                request.body.append('source', document.URL);
+              } else {
+                var body = new FormData();
+                // @ts-ignore
+                body.append('csrf_token', token);
+                body.append('source', document.URL);
```

Comment:
Should be able to reduce some of the duplication here? I.e. if request.body not instanceof FormData, set it to a new FormData and append the payload, then append the other two things outside the if/else?

Reply:
Done